vendor/github.com/status-im/ottoext/fetch/fetch.go

```diff
@@ -48,6 +49,9 @@ func (t *fetchTask) Execute(vm *otto.Otto, l *loop.Loop) error {
 		arguments = append(arguments, e)
 	}
 
+	vm.Lock()
+	defer vm.Unlock()
```

Comment:
Could you elaborate? As I can see, it's not used anywhere later in this function.

Reply:
Sure, that place was actually triggering race condition with fetch. A few lines down, there is a code: ` if _, err := h.Call("append", k, v); err != nil {` and `h.Call()` uses otto's VM underneath. `h` is a `otto.Value` object and there is a magic happening under the hood and it accesses VM directly. So the fastest and simplest solution was to guard the whole fetch Execute method with the same vm.Mutex.

Reply:
Oh, I see. Could you add a comment, please? That looks strange if you don't know reasons.

`l *loop.Loop` from the arguments is never used, by the way.

Reply:
Sure.

> l *loop.Loop from the arguments is never used, by the way.
Correct. Execute implements `Task` interface from `loop` package: https://github.com/status-im/status-go/blob/38d0ccc68eb5160e30cd051557bd60aada7b0fff/vendor/github.com/status-im/ottoext/loop/loop.go#L29

Reply:
Yeah, I saw it. The questions is do we still need it?

Reply:
Unless we refactor this package, it is needed, yes. I would be happy to get rid of this ambiguity once we settle on moving package to the `geth/jail` or keeping it as external dep.